addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxCubeBridgeHandler.java

```diff
@@ -147,21 +151,25 @@ public MaxCubeBridgeHandler(Bridge br) {
     private Socket socket = null;
     private BufferedReader reader = null;
     private OutputStreamWriter writer = null;
+    private final Lock socketLock = new ReentrantLock();
 
     private boolean previousOnline = false;
 
     private Set<DeviceStatusListener> deviceStatusListeners = new CopyOnWriteArraySet<>();
 
+    private final Lock pollingJobLock = new ReentrantLock();
```

Comment:
It was probably caused be another process eating up the threads in the scheduler. At first, I thought the scheduler was local to each binding (which seems more sensible to me).

I analysed by starting with the synchronised blocks. Since the class is already rather large and stateful, I decided to protect the shared variables separately. This gives fidelity on time-outs. These two variables were previously protected using a synchronized keyword. It would technically be possible for two external threads to start these jobs, both checking for nullability on the scheduled future and starting two separate jobs.